Document/0x05c-Reverse-Engineering-and-Tampering.md

```diff
@@ -1247,6 +1247,95 @@ The following approach can be used in order to patch the JavaScript file:
 
 #### Dynamic Instrumentation
 
+##### Information Gathering
+
+In this section we will learn about how to use Frida to obtain information about a running application.
+
+###### Getting Loaded Classes and their Methods
+
+You can use the command `Java` in the Frida CLI to access the Java runtime and retrieve information from the running app. Remember that, unlike Frida for iOS, in Android you need to wrap your code inside a `Java.perform` function. Thus, it's more convenient to use Frida scripts to e.g. get a list of loaded Java classes and their corresponding methods and fields or for more complex information gathering or instrumentation. One such scripts is listed below. The script to list class's methods used below is available on [Github](https://github.com/frida/frida-java-bridge/issues/44 "Github").
+
+```
+// Get list of loaded Java classes and methods
+
+//Filename: java_class_listing.js
+
+Java.perform(function() {
+    Java.enumerateLoadedClasses({
+        onMatch: function(className) {
+            console.log(className);
+            describeJavaClass(className);
+        },
+        onComplete: function() {}
+    });
+});
+
+// Get the methods and fields
+function describeJavaClass(className) {
+  var jClass = Java.use(className);
+  console.log(JSON.stringify({
+    _name: className,
+    _methods: Object.getOwnPropertyNames(jClass.__proto__).filter(function(m) {
+      return !m.startsWith('$') // filter out Frida related special properties
+        || m == 'class' || m == 'constructor' // optional
+    }),
+    _fields: jClass.class.getFields().map(function(f) {
+      return( f.toString());
+    })
+  }, null, 2));
+}
+
+//command
```

Comment:
```suggestion
Below the script is saved in the file java_class_listing.js and called with the flag `-l` and injected into the process ID specified.
```

Reply:
Can you please make a break after the script, put in a sentence like above and then the script is executed?

Reply:
unfortunately we cannot craft a suggestion for those changes because of the backticks involved. But please do so, I think it will look great :)